src/EnergyPlus/SimAirServingZones.cc

```diff
@@ -1123,33 +1121,43 @@ namespace SimAirServingZones {
 						PrimaryAirSystem( AirSysNum ).ControllerType( OASysControllerNum ) = ControllerType;
 						PrimaryAirSystem( AirSysNum ).ControlConverged( OASysControllerNum ) = false;
 						PrimaryAirSystem( AirSysNum ).CanBeLockedOutByEcono( OASysControllerNum ) = true;
-						//         Coil controllers can be entered either in the air loop controller list or the
-						//         OA system controller list. The CanBeLockedOutByEcono should only be set for OA coils
-						//         First get the OA controller actuator node and then compare to the air loop coil water inlet node
-						//         If these node numbers match, the coil is in the main air loop and the lockout flag should be reset to FALSE
 						GetControllerActuatorNodeNum( ControllerName, ActuatorNodeNum, errFlag );
-						for ( BranchNum = 1; BranchNum <= PrimaryAirSystem( AirSysNum ).NumBranches; ++BranchNum ) {
-							for ( CompNum = 1; CompNum <= PrimaryAirSystem( AirSysNum ).Branch( BranchNum ).TotalComponents; ++CompNum ) {
-								if ( SameString( PrimaryAirSystem( AirSysNum ).Branch( BranchNum ).Comp( CompNum ).TypeOf, "AirloopHVAC:OutdoorAirSystem" ) ) continue;
-								CompType = PrimaryAirSystem( AirSysNum ).Branch( BranchNum ).Comp( CompNum ).TypeOf;
-								WaterCoilNodeNum = -1;
-								if ( SameString( CompType, "Coil:Cooling:Water:DetailedGeometry" ) || SameString( CompType, "Coil:Heating:Water" ) || SameString( CompType, "Coil:Cooling:Water" ) ) {
-									WaterCoilNodeNum = GetCoilWaterInletNode( PrimaryAirSystem( AirSysNum ).Branch( BranchNum ).Comp( CompNum ).TypeOf, PrimaryAirSystem( AirSysNum ).Branch( BranchNum ).Comp( CompNum ).Name, ErrorsFound );
-								}
-								if ( WaterCoilNodeNum == ActuatorNodeNum ) {
-									PrimaryAirSystem( AirSysNum ).CanBeLockedOutByEcono( OASysControllerNum ) = false;
+
+						bool nonLockoutCoilFound = false;
+						WaterCoilNodeNum = -1;
+						// added to fix bug issue #5695, if HW coil on outdoor air system, don't lock out during economizing
+						if ( OANum > 0 ) {
+							for ( int OACompNum = 1; OACompNum <= OutsideAirSys( OANum ).NumComponents; ++OACompNum ) {
+								CompType = OutsideAirSys( OANum ).ComponentType( OACompNum );
+								if ( SameString( CompType, "Coil:Heating:Water" ) ) {
```

Comment:
OK, this caught me off guard, but I guess this is just inside getinput.  And there's no chance the controls landscape could change *within* a simulation that would necessitate rechecking this; right?

Reply:
And the below switch checks for several coil types, while this only checks one.  I guess that's because this is purely heating.  OK.